### ISSUE TITLE
`relative-url-style` turns `'./'` into `''`, which breaks things

### ISSUE THREAD

**Issue by wooorm** [sindresorhus/eslint-plugin-unicorn#1747]
The JavaScript code:

```js
[new URL('./', 'https://example.com/a/b/c.html').href, new URL('.', 'https://example.com/a/b/c.html').href, new URL('', 'https://example.com/a/b/c.html').href]
```

Yields

```
["https://example.com/a/b/", "https://example.com/a/b/", "https://example.com/a/b/c.html"]
```

The new `relative-url-style` rule in xo turns `'./'` into `''`, which is different, and changes code. I believe it should turn `'./'` into `'.'` instead?

Related-to: GH-1672.

Found in the wild here: https://github.com/mdx-js/mdx/blob/656a4ae5e16989f0fc87a053a0e07b1a2b253a5d/packages/loader/test/index.test.js#L23

**Comment by fisker** [comment]
Thanks for the report.
